sisl/viz/plotly/plots/pdos.py

```diff
@@ -313,11 +326,14 @@ def _after_read(self):
         """
         from xarray import DataArray
 
-        # Normalize self.PDOS to do the same treatment for both spin-polarized and spinless simulations
-        self.spin_polarized = len(self.PDOS.shape) == 3
+        # Check if the PDOS contains spin resolution (there should be three dimensions,
+        # and the first one should be the spin components)
+        self.spin_polarized = self.PDOS.squeeze().ndim == 3
```

Comment:
Are you also checking for non-colinear?
The last dimension may be either 1, 2 or 4?

Reply:
Do you have an example of a non-colinear PDOS file? What are each of the 4 spin channels in the non-colinear case?

Reply:
See [here](https://github.com/zerothi/sisl/blob/293969f45ef21af2d63a04ca60a7dda5ab673008/sisl/io/siesta/pdos.py#L102)
and the following lines.

Reply:
Aah ok, is the sum channel just the sum of x, y and z? Or is it more complex?

Reply:
No, it is more complex. The total DOS (sum) may be 4, while all the rest may be 0. You can think of it like a polarized calculation (here x and y are always 0). For a spin polarized up=1, down=1 one would have sum=2 and z = up - down = 0.
For up=1, down=0.5 one would have sum=1.5 and z=0.5.

So it is the *net* spin of that orbital. However one will always have `|x**2+y**2+z**2|**0.5 <= sum`.

Reply:
Ok, thanks! I have a non-colinear PDOS file and I'm trying to see what I need to change to make it work now :)

Reply:
> By the way, I didn't use your `read_data(as_dataarray=True)` because it seemed like there wasn't an atom index, so I didn't understand. For two orbitals that have the same parameters but are in a different atom, is the DOS information summed?

No, true there isn't. But that could readily be added.
```python
a = sile.read_data(as_dataarray=True)
geom = a.attrs.geometry
atom_idx = geom.o2a(np.arange(geom.no))
```
done.
I would actually like to reduce number of coordinates since the stored matrix in `xarray` is a dense matrix. So adding a new coordinate needs careful thinking.
I am